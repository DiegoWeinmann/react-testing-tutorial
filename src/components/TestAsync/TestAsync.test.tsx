import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
  waitForElement
} from '@testing-library/react';
import TestAsync from './TestAsync';

const renderTestAsync = () => {
  return render(<TestAsync />);
};

afterEach(cleanup);

describe('<TestAsync/>', () => {
  it('matches the snapshot', () => {
    const { asFragment } = renderTestAsync();
    expect(asFragment()).toMatchSnapshot();
  });

  it.only('increments counter after .5s', async () => {
    const { getByTestId, getByText } = renderTestAsync();

    fireEvent.click(getByTestId('button-up'));
    const counter = await waitForElement(() => getByText('1'));
    expect(counter).toHaveTextContent('1');
  });
});
