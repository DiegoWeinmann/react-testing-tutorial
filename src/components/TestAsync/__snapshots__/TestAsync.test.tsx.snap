// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestAsync/> matches the snapshot 1`] = `
<DocumentFragment>
  <h1
    data-testid="counter"
  >
    0
  </h1>
  <button
    data-testid="button-up"
  >
     Up
  </button>
  <button
    data-testid="button-down"
  >
    Down
  </button>
</DocumentFragment>
`;
